x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/aggregation/RateLongGroupingAggregatorFunction.java

```diff
@@ -571,36 +573,42 @@ public void evaluateFinal(Block[] blocks, int offset, IntVector selected, Groupi
         BlockFactory blockFactory = driverContext.blockFactory();
         int positionCount = selected.getPositionCount();
         try (var rates = blockFactory.newDoubleBlockBuilder(positionCount)) {
+            Map<Integer, ReducedState> flushedStates = new HashMap<>(positionCount);
```

Comment:
- What do we get in the `selected` int vector? Does include all ts buckets in this shard for a query?
- If this is the case then this `Map` can grow significantly? Should we use something else? `LongObjectPagedHashMap`?